### PR TITLE
[READY] Stacks will change item_states and w_class based on how full they are.

### PR DESCRIPTION
**Pull request by Expletive**
:cl:
add:  Many stacks now update their sprite based on their amount.
add: Stacks will now weigh less if they're less than full.
imageadd: Added new icon states for glass, reinforced glass, metal, plasteel, plastic, plasma, plastitanium, titanium, gold, silver, adamantine, brass, bruise packs, ointment, gauze, cloth, leather, wet leather, hairless hide, human hide, ash drake hide, goliath hide, bones, sandstone blocks, and snow blocks.
tweak: Some lavaland stacks' max amounts have been reduced.
/:cl:

This will make it much easier to tell at a glance how many sheets of a thing are in a given stack. It will be especially helpful medicine, as they naturally have much lower stack sizes than other stacks.  The sprites and w_class will both be adjusted at 1/3rd and 2/3rds of the stack's max_amount. 

mining
items

### CODE REVIEW

**Comment by RemieRichards** [comment]
This is nice, but I feel "sheet" ones should be closer together, like leather is in the image (but still vertical obviously)

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 1]
```diff
@@ -12,12 +12,17 @@
...
 	var/mineralType = null
+	novariants = TRUE
 
 /obj/item/stack/tile/Initialize(mapload, amount)
 	. = ..()
 	pixel_x = rand(-3, 3)
 	pixel_y = rand(-3, 3) //randomize a little
 
+/obj/item/stack/tile/update_icon()
+	..(TRUE)
```
Comment (RemieRichards) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: this true isn't used/going anywhere?

Comment (Expletive) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: Oh, right. Forgot about that. I'll get rid of it.

**Comment by Expletive** [comment]
@RemieRichards So you want the sheet types to be like this?
titanium

**Comment by RemieRichards** [comment]
Yes, MUCH better.

**Comment by Expletive** [comment]
And I didn't really understand what you were trying to say about plasma. How do you think it should be?

**Comment by RemieRichards** [comment]
Like, normally plasma is jagged, but when you've lined them up they form a square almost and it looks bad.
So what I'm saying is for each plasma, if it's an even number, shift it up slightly
eg:
1 plasma, not affected
2 plasma, 1 is normal, 2 is shifted up
3 plasma, 1 is normal, 2 is shifted up, 3 is normal
4 plasma, 1 is normal, 2 is shifted up, 3 is normal, 4 is shifted up
etc.

I'm not sure how it'll look, but I believe it'll help alleviate the square-ness.

**Comment by Expletive** [comment]
Oh, yeah. I tried that, but the plasma sprites don't have outlines, so they don't really stack too well. With the sprites in the OP, I was going for a "A somewhat larger chunk of plasma" sort of look.

**Comment by RemieRichards** [comment]
perhaps make them spikier then, if you want that route, anything but square tbh.

**Comment by Expletive** [comment]
untitled

Okay, how is this?

**Comment by RemieRichards** [comment]
better, yes

**Comment by WJohn** [comment]
Fancy as hell! I like this!

**Comment by XDTM** [comment]
I love this, nice job

**Comment by GunHog** [comment]
Wonderful idea!

**Comment by Expletive** [comment]
lavaland
Some possibilities for the lavaland stacks. What do you think @WJohn

**Comment by octareenroon91** [comment]
Are the ointment tube sprites the right way around?

**Comment by Expletive** [comment]
They are. They've always been oriented that way @octareenroon91

**Comment by Chronitonity** [comment]
Plasma still looks a bit weird but its probably unavoidable without changing the sprite

**Review by Cyberboss — COMMENTED** [review]
:+1: Looks fine, but I haven't checked the dmis

**Comment by octareenroon91** [comment]
No, I mean the ointment tubes go from full to empty, unlike all the other sprite sequences. Can you confirm they're labeled the right way around in the dmi?

**Comment by Bawhoppen** [comment]
solid PR

**Comment by RandomMarine** [comment]
Among other physical feats such as casually carrying hundreds of thousands of cubic centimeters of solid steel and molding it into comfy beds with their bare hands. Spacemen can now add squeezing one half-tube of ointment into another to the list.

**Comment by Bawhoppen** [comment]
they always could, but just couldn't see it

**Pull request merged**